src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -552,6 +558,12 @@ case class WorkflowActor(workflow: WorkflowDescriptor, backend: Backend)
     context.system.scheduler.scheduleOnce(ResendInterval, self, message)
   }
 
+  private def removePendingCallKeyPersistence(data: WorkflowData, message: PersistenceMessage) = {
+    logger.debug(s"Got whenUnhandled message: $message")
+    val updatedData = data.removePersisting(message.callKey, message.executionStatus)
+    stay using updatedData
```

Comment:
It needs to return the next state + state data, though it would be nice to see an explicit return type on the method.